drivers/storage/portworx/component/autopilot.go

```diff
@@ -265,26 +305,138 @@ func (c *autopilot) createConfigMap(
+					Namespace: namespace,
+				},
+			},
+		},
+	)
+}
+
 func (c *autopilot) createSecret(clusterNamespace string, ownerRef *metav1.OwnerReference) error {
 
-	token, cert, err := c.getPrometheusTokenAndCert()
+	ocp416Plus, err := pxutil.IsSupportedOCPVersion(c.k8sClient, pxutil.Openshift_4_16_version)
 	if err != nil {
 		return err
 	}
 
+	secret := &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            AutopilotSecretName,
+			Namespace:       clusterNamespace,
+			OwnerReferences: []metav1.OwnerReference{*ownerRef},
+		},
+	}
+
+	if ocp416Plus {
+		err = k8sutil.GetSecret(c.k8sClient, AutopilotSecretName, clusterNamespace, secret)
+		if err != nil {
+			return fmt.Errorf("Error during getting secret %v ", err)
+		}
+
+		if secret.Data != nil {
+			// if secret exists, check secret type
+			if secret.Type == v1.SecretTypeServiceAccountToken {
+				// if secret type is service account token, check if token is expired
+				ok, err := isTokenRefreshRequired(secret)
+				if err != nil {
+					return fmt.Errorf("Error during checking token refresh %v ", err)
+				}
+				if ok {
+					// refresh the token if it is expired
+					logrus.Infof("refreshing token %s in namespace %s", AutopilotSecretName, clusterNamespace)
+					err := c.refreshTokenSecret(secret, clusterNamespace, ownerRef)
+					if err != nil {
+						return fmt.Errorf("Error during refreshing token %v ", err)
+					}
+
+					// delete autopilot deployment to sync with new token
+					logrus.Info("deleting autopilot pod to sync with new token")
+					err = k8sutil.DeletePodsByLabel(c.k8sClient, map[string]string{"name": "autopilot"}, clusterNamespace)
+					if err != nil {
+						return fmt.Errorf("Error during deleting deployment %v ", err)
+					}
+					logrus.Infof("token refreshed successfully")
+				}
+				return nil
+			} else {
+				// if secret type is not service account token, delete the secret
+				// this is to handle first time creation of secret on OCP 4.16+
+				err := k8sutil.DeleteSecret(c.k8sClient, AutopilotSecretName, clusterNamespace, *ownerRef)
+				logrus.Infof("Deleted secret to recreate %s in namespace %s", AutopilotSecretName, clusterNamespace)
+				if err != nil {
+					return fmt.Errorf("Error during deleting secret %v ", err)
+				}
+			}
+		}
+
+		// if secret is not created, create the secret
+		// create secret with service account token
+		secret.Type = v1.SecretTypeServiceAccountToken
+		secret.Annotations = map[string]string{
+			"kubernetes.io/service-account.name": AutopilotPrometheusServiceAccountName,
+		}
+
+		token, err := generateAPSaToken(clusterNamespace, defaultAutoPilotSaTokenExpirationSeconds)
+		if err != nil {
+			return fmt.Errorf("Error during generating token %v ", err)
+		}
+		secret.Data = make(map[string][]byte)
+		secret.Data["token"] = token
+		secret.Data[AutopilotSaTokenRefreshTimeKey] = []byte(time.Now().UTC().Add(time.Duration(defaultAutoPilotSaTokenExpirationSeconds/2) * time.Second).Format(time.RFC3339))
```

Comment:
we need to read the actual token expiration duration instead of relying on `defaultAutoPilotSaTokenExpirationSeconds`. See my comment below.